lib/_http_outgoing.js

```diff
@@ -1005,6 +1026,131 @@ function connectionCorkNT(conn) {
   conn.uncork();
 }
 
+OutgoingMessage.prototype.writeFile = function writeFile(file, opts, callback) {
+  if (this[kLocked]) {
+    throw new ERR_STREAM_LOCKED('writeFile');
+  }
+
+  let start;
+  let end = Infinity;
+
+  if (typeof opts === 'function') {
+    callback = opts;
+    opts = null;
+  } else if (opts != null) {
+    validateObject(opts, 'opts');
+
+    start = opts.start;
+    if (start !== undefined) {
+      validateInteger(start, 'opts.start', 0);
+    }
+
+    end = opts.end;
+    if (end === undefined) {
+      end = Infinity;
+    } else if (end !== Infinity) {
+      validateInteger(end, 'opts.end', 0);
+
+      if (start !== undefined && start > end) {
+        throw new ERR_OUT_OF_RANGE('start', `<= "end" (here: ${end})`, start);
+      }
+    }
+
+    validateFunction(callback, 'callback');
+  }
+
+  try {
+    this[kLocked] = true;
+    if (typeof file === 'number') {
```

Comment:
Can you add a comment to here and say that: `// File is a number if it is a file descriptor`